### ISSUE TITLE
Websocket checkStartsWith doesn't work as expected

### ISSUE THREAD

**Issue by travishaagen** [netty/netty#10782]
### Expected behavior

The websocket `checkStartsWith` (#10583) feature should support all of the following URLs:

- /
- /?param
- /subPath
- /subPath?param

### Actual behavior

No matter whether you set it to true/false it only supports the websocket path itself with:

- /
- /?param

### Steps to reproduce

Modify the websocket server example to set this config param?

### Minimal yet complete reproducer code (or URL to code)

https://github.com/netty/netty/blob/4.1/example/src/main/java/io/netty/example/http/websocketx/server/WebSocketServerInitializer.java

change applicable line to:

```
pipeline.addLast(new WebSocketServerProtocolHandler(WEBSOCKET_PATH, null, true, 65555, false, true));

// and also test

pipeline.addLast(new WebSocketServerProtocolHandler(WEBSOCKET_PATH, null, true, 65555, false, false));
```

### Netty version

4.1.53.Final

### JVM version (e.g. `java -version`)

JDK 11

### OS version (e.g. `uname -a`)

10.14.x

**Comment by travishaagen** [comment]
As a workaround, I seem to be able to overwrite the URI before the handshake: `req.setUri("/");`

**Comment by chrisvest** [comment]
@Doyuni Can you take a look?

**Comment by Doyuni** [comment]
@chrisvest Ok, I'll check this.

**Comment by Doyuni** [comment]
@travishaagen 

I understand that if set websocket path "/test", expect below URLs to pass right?
- /test
- /test?param
- /test/subPath
- /test/subPath?param

**Comment by travishaagen** [comment]
@Doyuni : Yes. I haven't tested the latest Netty release either. Thx :)

**Comment by Doyuni** [comment]
@travishaagen 
I wrote test code and ran the tests on the above example. All tests were performed normally.
You can test them with the latest version.
Thanks :)

**Comment by normanmaurer** [comment]
So I am correct that we can close this one ?

**Comment by travishaagen** [comment]
Apologies for the delay. Please give me a chance to verify on Monday, after the US holiday.

**Comment by travishaagen** [comment]
Verified fix with `4.1.54.Final` release. Thank you for taking a look @Doyuni
